src/events.ts

```diff
@@ -0,0 +1,48 @@
+import { EmojiActor } from "./actor";
+
+/**
+ * Handler for a user interaction with an individual emoji.
+ *
+ * @param actor   Actor being interacted with.
+ */
+export type EmojiEventHandler = (actor: EmojiActor) => void;
+
+/**
+ * Handlers for user interactions with individual emojis.
+ */
+export interface EmojiEvents {
+	/**
+	 * Handler for a user clicking an emoji.
+	 */
+	onClick: EmojiEventHandler;
+}
+
+/**
+ * DOM attribute indicating that events were initialized for a container.
+ */
+const attributeIndicator = "data-emojisplosion-events-initialized";
+
+const domNodesToActors = new WeakMap<EventTarget, EmojiActor>();
+
+export function initializeEvents(
+	actors: EmojiActor[],
+	container: Element,
+	events: EmojiEvents
+) {
+	for (const actor of actors) {
+		domNodesToActors.set(actor.element, actor);
+	}
+
+	if (container.hasAttribute(attributeIndicator)) {
+		return;
+	}
+
+	container.setAttribute(attributeIndicator, "true");
+
+	container.addEventListener("click", (event) => {
+		const actor = event.target && domNodesToActors.get(event.target);
+		if (actor) {
+			events.onClick(actor);
```

Comment:
I wonder if it's worth passing through the original event to the handler as well? e.g. that could give you the position of the click.

Though I did poke around with trying to use the position of the click to update the x velocity, but it doesn't seem accurate enough to look right (possibly due to timing, I didn't dig too deep), so maybe it's not that useful 🤷

Reply:
Sorry for the delay here - was traveling. But yes that makes sense! Just on instinct I'll change it to `({ actor, event })`.